src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -582,31 +601,82 @@ void QgsSymbolLegendNode::updateLabel()
 
   bool showFeatureCount = mLayerNode->customProperty( QStringLiteral( "showFeatureCount" ), 0 ).toBool();
   QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( mLayerNode->layer() );
+  mLabel = symbolLabel();
 
-  if ( mEmbeddedInParent )
+  if ( showFeatureCount && vl )
   {
-    QString layerName = mLayerNode->name();
-    if ( !mLayerNode->customProperty( QStringLiteral( "legend/title-label" ) ).isNull() )
-      layerName = mLayerNode->customProperty( QStringLiteral( "legend/title-label" ) ).toString();
+    qlonglong count = mEmbeddedInParent ? vl->featureCount() : vl->featureCount( mItem.ruleKey() ) ;
+    mLabel += QStringLiteral( " [%1]" ).arg( count != -1 ? QLocale().toString( count ) : tr( "N/A" ) );
+  }
+
+  emit dataChanged();
+}
+
+QString QgsSymbolLegendNode::evaluateLabel( QgsExpressionContext context, const QString label )
```

Comment:
Both parameters should be const references. Please review the whole pull request for such parameters and make sure they are always const references!

Reply:
Yeah I tried it because I thought this was an oversight on my end, but if you check 626 and 640, you can see why it cannot be a const.

Trust me I have checked all the params already.

Reply:
* The QString is not a reference
* The context should also be a const reference and an internal copy taken if you need to modify

Reply:
Done.